src/filler/mod.rs

```diff
@@ -0,0 +1,186 @@
+//! A module containing default values filler.
+use serde_json::{json, Value};
+
+use crate::schema::{PrimitiveType, Schema};
+
+// Recursively check if the object is empty
+//
+// If the child object is empty, parent object is considered as empty.
+fn is_empty_object(value: &Value) -> bool {
+    match value {
+        Value::Object(map) => map.values().all(|x| is_empty_object(x)),
+        _ => false,
+    }
+}
+
+fn fill_defaults(schema: &Schema, data: &mut Value, include_optional: bool) {
+    match (schema.r#type().primitive_type(), schema.r#type().is_required()) {
+        (PrimitiveType::Object, _) => {
```

Comment:
I found it generally useful for readability to separate the control flow (so e.g. the match statement here) from the details of what happens in particular branch. What do you think about extracting methods here - one per match branch for example ?